onnxruntime/core/session/environment.cc

```diff
@@ -48,6 +49,23 @@ Status Environment::Create(std::unique_ptr<logging::LoggingManager> logging_mana
   return status;
 }
 
+Status Environment::RegisterSharedAllocator(OrtAllocator* allocator) {
+  const auto* mem_info_ptr = allocator->Info(allocator);
```

Comment:
If we combine create+register this can probably take a shared_ptr to make the ownership semantics clearer.